React/Modules/RCTUIManager.m

```diff
@@ -793,10 +793,10 @@ - (void)_removeChildren:(NSArray<UIView *> *)children
     // original `superview` (which can differ from `container`) and an index of removed view.
     UIView *originalSuperview = removedChild.superview;
     NSUInteger originalIndex = [originalSuperview.subviews indexOfObjectIdenticalTo:removedChild];
-    [container removeReactSubview:removedChild];
     // Disable user interaction while the view is animating
     // since the view is (conceptually) deleted and not supposed to be interactive.
     removedChild.userInteractionEnabled = NO;
+    [container removeReactSubview:removedChild];
```

Comment:
Why you had to move this?

Reply:
That's because `removeReactSuperview` contains `[subview removeFromSuperview]` call which eventually fires `RCTModalHostView` lifecycle methods (`didMoveToWindow`, `didMoveToSuperview`) and it checks if it's `userInteractionEnabled` is disabled before actually disabling it here